graphix/pattern.py

```diff
@@ -97,6 +104,9 @@ def __init__(self, input_nodes: list[int] | None = None) -> None:
         # output nodes are initially input nodes, since none are measured yet
         self.__output_nodes = list(input_nodes)
 
+        if cmds:
```

Comment:
`Iterable` may not implement `__bool__` .

```suggestion
        cmds = list(cmds)
        if cmds:
```

Reply:
Indeed, thank you for reporting this. Fixed in 74d5a5d. I preferred to check that `cmds is not None` instead of instantiating a list.